Panic with some go file

**Describe the bug**

Run `scc` with this file `https://github.com/360EntSecGroup-Skylar/goreporter/blob/master/engine/template.html.go`, and you will get the panic

```
panic: runtime error: index out of range

goroutine 23 [running]:
github.com/boyter/scc/processor.CountStats(0xc000254000)
```

**To Reproduce**

1. Tell us how you were using `scc` either as command line or library.

Both.

2. If command line please list the arguments. If library please supply code to demonstrate the issue.


3. If the issue is a miscount or some such please link or supply a file which demonstrates the issue.

```
https://github.com/360EntSecGroup-Skylar/goreporter/blob/master/engine/template.html.go
```

**Expected behavior**

**Desktop (please complete the following information):**
 - OS:macOS
 - Version [e.g. 22]

--- a/processor/workers.go
+++ b/processor/workers.go
@@ -124,7 +124,7 @@ func stringState(fileJob *FileJob, index int, endPoint int, stringTrie *Trie, en
 
 		// If we are in a literal string we want to ignore the \ check OR we aren't checking for special ones
 		if ignoreEscape || fileJob.Content[i-1] != '\\' {
-			if ok, _, _ := stringTrie.Match(fileJob.Content[i:]); ok != 0 {
+			if checkForMatchSingle(fileJob.Content[i], index, endPoint, endString, fileJob) {
 				return i, SCode
 			}
 		}

--- a/processor/workers.go
+++ b/processor/workers.go
@@ -469,6 +469,13 @@ func CountStats(fileJob *FileJob) {
 			}
 		}
 
+		// We shouldn't normally need this, but unclosed strings or comments
+		// might leave the index past the end of the file when we reach this
+		// point.
+		if index >= len(fileJob.Content) {
+			return
+		}
+
 		// Only check the first 10000 characters for null bytes indicating a binary file
 		// and if we find it then we return otherwise carry on and ignore binary markers
 		if index < 10000 && fileJob.Binary {

--- a/processor/workers_test.go
+++ b/processor/workers_test.go
@@ -981,6 +981,17 @@ code
 	}
 }
 
+func TestCountStatsIssue106(t *testing.T) {
+	ProcessConstants()
+	fileJob := FileJob{
+		Language: "Go",
+	}
+
+	fileJob.Content = []byte("foo = `\nabc\"\ndef\n`")
+
+	CountStats(&fileJob)
+}
+
 //////////////////////////////////////////////////
 // Benchmarks Below
 //////////////////////////////////////////////////

